packages/components/src/internal/util/Date.ts

```diff
@@ -71,11 +79,26 @@ export function getColDateFormat(queryColumn: QueryColumn, dateFormat?: string,
     // Issue 44011: account for the shortcut values (i.e. "Date", "DateTime", and "Time")
     if (rawFormat === 'Date') rawFormat = getMomentDateFormat();
     if (rawFormat === 'DateTime') rawFormat = getMomentDateTimeFormat();
-    if (rawFormat === 'Time') rawFormat = getTimeFormat();
+    if (rawFormat === 'Time') rawFormat = getMomentTimeFormat();
 
     // Moment.js and react datepicker date format is different
     // https://github.com/Hacker0x01/react-datepicker/issues/1609
-    return rawFormat.replace('YYYY', 'yyyy').replace('YY', 'yy').replace('DD', 'dd');
+    return momentToDateFNS(rawFormat);
+}
+
+// Issue 48300: Respect 12-hour vs 24-hour time display format
+// This method intends to parse out the time portion of the format from the date portion of the format.
+// NK: That said, this is a far-reaching over simplification / contrived implementation which presumes the time
+// format follows the date format. For a more precise implementation we would search for time-specific portions within
+// the string (or use some more grand date format parsing library utility), however, there are so many different
+// formats (Java, Moment, Date-FNS, JavaScript, etc) and a seemingly infinite number of ways to configure a date/time
+// format that I've elected to just assume the second part of a space-split string that contains a ":" is the time
```

Comment:
This seems like a reasonable compromise to make. I think most datetime formats have the time in the second part of the format, even in european dates. I haven't seen a datetime format have the time be first.